src/lib/mina_generators/parties_generators.ml

```diff
@@ -1101,17 +1123,25 @@ let gen_parties_from ?failure ?(max_other_parties = max_other_parties)
   (* table of public keys not in the ledger, to be used for new parties
      we have the corresponding private keys, so we can create signatures for those new parties
   *)
-  let ledger_account_set =
+  let ledger_account_list =
     Account_id.Set.union_list
       [ Ledger.accounts ledger
       ; Account_id.Set.of_hashtbl_keys account_state_tbl
       ]
+    |> Account_id.Set.to_list
+  in
+  let ledger_pk_list =
+    List.map ledger_account_list ~f:(fun account_id ->
+        Account_id.public_key account_id )
+  in
+  let ledger_pk_set =
+    Signature_lib.Public_key.Compressed.Set.of_list ledger_pk_list
   in
   let available_public_keys =
```

Comment:
should we track available account ids here? It could use the same key to create a new account with a different token ID, right?

Reply:
That's right. But since we only generate new account with fresh keypairs, so I filter out the `available_public_keys` using only keys instead of account_ids.